examples/basecoin/cmd/basecoind/main.go

```diff
@@ -1,27 +1,74 @@
 package main
 
 import (
+	"encoding/json"
 	"fmt"
 	"os"
 
+	"github.com/spf13/cobra"
+
+	"github.com/tendermint/tmlibs/cli"
 	dbm "github.com/tendermint/tmlibs/db"
 	"github.com/tendermint/tmlibs/log"
 
-	"github.com/cosmos/cosmos-sdk/baseapp"
 	"github.com/cosmos/cosmos-sdk/examples/basecoin/app"
+	"github.com/cosmos/cosmos-sdk/server"
+	"github.com/cosmos/cosmos-sdk/version"
+)
+
+// gaiadCmd is the entry point for this binary
+var (
+	gaiadCmd = &cobra.Command{
+		Use:   "gaiad",
+		Short: "Gaia Daemon (server)",
+	}
 )
 
+// defaultOptions sets up the app_options for the
+// default genesis file
+func defaultOptions(args []string) (json.RawMessage, error) {
+	addr, secret, err := server.GenerateCoinKey()
+	if err != nil {
+		return nil, err
+	}
+	fmt.Println("Secret phrase to access coins:")
+	fmt.Println(secret)
+
+	opts := fmt.Sprintf(`{
+      "accounts": [{
+        "address": "%s",
+        "coins": [
+          {
+            "denom": "mycoin",
+            "amount": 9007199254740992
+          }
+        ]
+      }]
+    }`, addr)
+	return json.RawMessage(opts), nil
+}
+
 func main() {
-	fmt.Println("This is temporary, for unblocking our build process.")
-	return
+	// TODO: this should somehow be updated on cli flags?
+	// But we need to create the app first... hmmm.....
```

Comment:
cant we just let StartCmd handle all this? its the only one that needs the DB right

Reply:
But to be generic, StartCmd takes a prepared app.
Otherwise, StartCmd contains app-specific setup code

Reply:
I guess we can pass in a function to StartCmd that returns a abci.Application, to init lazily?

Reply:
Just did this and pushed it